src/types.mts

```diff
@@ -0,0 +1,112 @@
+// Local Types
+type TConfigurationObjectValue = string | RegExp | object | boolean | typeof Function;
+
+type TCustomConfiguration = Omit<IConfigurationInternal, 'accessToken'> & {
+  onUnhandledError?: string;
+  onUnhandledPromiseRejection?: string;
+};
+
+type TSerializedObject<T extends object> = {
+  // eslint-disable-next-line @typescript-eslint/ban-types -- any function can be expected here
+  [key in keyof T]: T[key] extends Function | RegExp ? string : T[key];
+};
+
+type TSubmitterParameters = [
+  string,
```

Comment:
Great idea, I've added the log levels as the first argument to the type. When I try instantiating a new `RollbarClient` instance and calling `client.log()`, the first argument gets auto-filled with the five possible log levels. Is that what you meant?

Reply:
So that's what I meant with the very first part of this comment, but this still doesn't address the second part.

I believe at the moment, when you type `client.log()`, your autocomplete will give you the type of each parameter the function takes, but they will be unnamed, so it's a bit hard as a consumer to know what should go into it (I must read the docs).

If you type the function regularly with each parameter named, then the editor will also give consumers of the library the name of each parameter in addition to their types, which should help quite a bit.

Reply:
Oh i got you now, that IS a good idea. I've added it and indexed `TSubmitterParameters` for each parameter so they stay in sync. Thank you for the suggestion!